lib/spack/spack/version.py

```diff
@@ -146,6 +146,10 @@ def underscored(self):
     def dashed(self):
         return '-'.join(str(x) for x in self.version)
 
+    @property
+    def joined(self):
```

Comment:
a unit test would be nice to have to make sure the behaviour won't change down the road